### ISSUE TITLE
Dropdown menu have a <div> directly inside a <ul>

### ISSUE THREAD

**Issue by nloureiro** [ethereum/ethereum-org-website#8500]
### Describe the bug

Dropdown menu have a <div> directly inside a <ul> and it's not valid html

inside a <ul> needs to be a <li>

### To reproduce

1. Go to ant page
2. Click on "Learn" form the top navigation on the header
3. inspect the markup from the dropdown options
4. confirm what it's on the screenshot

### Expected behavior

should ve <ul> <li> <div> << if needed to have the same outcome

### Screenshots

![Screen Shot 2022-11-04 02 47 39 PM](https://user-images.githubusercontent.com/1120748/200005959-92feff06-6b93-41c3-a581-04402340a762.png)

**Comment by TylerAPfledderer** [comment]
Hey @nloureiro! Indeed, per W3 content model spec, only `li` tags should be direct descendants of the `ul` element.

I would argue that an improvement would be to have separate sub-lists with each heading tied to the sublist (via `aria-labelledby` say). To keep the navigation content dynamic, maybe use logic based on the structure of the [linkSections](https://github.com/TylerAPfledderer/ethereum-org-website/blob/dc15c137893850f825386b089db877497833aeb4/src/components/Nav/index.tsx#L150) object if sub lists need to be created. The `Learn` menu is the only one with this issue though.

Would have to wait until the next wave of components are available for migration (#6374) for someone to tackle this (or after), but what does @pettinarip think about this and my thoughts above?

**Comment by nloureiro** [comment]
> Hey @nloureiro! Indeed, per W3 content model spec, only `li` tags should be direct descendants of the `ul` element.
> 
> I would argue that an improvement would be to have separate sub-lists with each heading tied to the sublist (via `aria-labelledby` say). To keep the navigation content dynamic, maybe use logic based on the structure of the [linkSections](https://github.com/TylerAPfledderer/ethereum-org-website/blob/dc15c137893850f825386b089db877497833aeb4/src/components/Nav/index.tsx#L150) object if sub lists need to be created. The `Learn` menu is the only one with this issue though.
> 
> Would have to wait until the next wave of components are available for migration (#6374) for someone to tackle this (or after), but what does @pettinarip think about this and my thoughts above?

Yes, it might be better to wait for the next wave of components from the UI migration

**Comment by minimalsm** [comment]
This isn't broken (tested on screenreaders too), but I agree we should endeavour to use valid HTML :-) Removed bug tag and added a11y tag instead.

**Comment by TylerAPfledderer** [comment]
> This isn't broken (tested on screenreaders too), but I agree we should endeavour to use valid HTML :-) Removed bug tag and added a11y tag instead. 

Correct @minimalsm. No problems for me when using NVDA. :)

**Comment by pettinarip** [comment]
Correct. I think we should have a `ul` for each sublist.

Maybe something like this 🤔 
```html
<li>
  <!-- define the two columns layout first! -->

  <!-- column 1 -->
  <div>
    <h2>Sublist title</h2>
    <ul>
      <li></li>
      <li></li>
      ...
    </ul>

    <h2>Sublist title 2</h2>
    <ul>
      <li></li>
      <li></li>
      ...
    </ul>

    <h2>Sublist title 3</h2>
    <ul>
      <li></li>
      <li></li>
      ...
    </ul>
  </div>

  <!-- column 2 -->
  <div>
    <h2>Sublist title 4</h2>
    <ul>
      <li></li>
      <li></li>
      ...
    </ul>

    <h2>Sublist title 5</h2>
    <ul>
      <li></li>
      <li></li>
      ...
    </ul>
  </div>
</li>
```

In terms of timing, the `Nav` is not listed yet on the migration to Chakra. So...if someone wants to tackle this now or in the following weeks, I would say let's do it.

**Note**: I'll block this issue if I see that we are migrating it and no one has picked it yet.

**Comment by TylerAPfledderer** [comment]
@pettinarip circling back to this. Not stale!

I believe your example is appropriate because all of this is inside the upper `li` where the "Learn" text is, and each subsection is it's own list with clear definitions stating such.

Another thought I have is whether each sublist should be using a heading tag and instead use simply a styled `span`? I've never seen an accessible site use an h2, when most recommendations I see suggest headings are more for regions of content. The label can be tied to the lists too via `aria-labelledby`.

I would think headings for sublists in navigation are closer to simpler labels like in forms than denoting larger chunks of content, article briefs, etc. Or the heading be tied to the overall navigation region if not the `aria-label`.

This could purely be a semantic choice, but it does affect how a user navigates through headings on a screen reader when first visiting a page (they might expect main page content?)

What do you think? 😄

**Comment by pettinarip** [comment]
@TylerAPfledderer 

I don't have a strong opinion about using an `h2` vs `span` but I agree with you on your thinking about the purpose the heading should have. Meaning that you convince me and we should use a `span` :)

Not sure though if we must have the `aria` attrs. Do you think that with the `span` we should still add those attrs?

In this example they are not using any aria attr <https://www.w3.org/WAI/tutorials/page-structure/content/#nested-lists> but haven't test it myself.

**Comment by TylerAPfledderer** [comment]
@pettinarip I guess you don't have to here! With `aria-labelledby` that was a consideration regardless of the element used for the label.

This is its own discussion in properly handling navigation in relation to a screen reader and more so with nested lists of links. It could come down to how the user with the screen reader might approach navigating through the many labels and links. 

**In short** we can leave it to simply the improved structure of the HTML. :) User feedback would be needed if there was indeed an issue with screen reader use that would prompt further modification with aria-attributes. 

At least having the current main items (i.e. "Learn") declared with button-like functionality and `aria-expanded` are a great help!

> There are a few articles I've found that discussed the complexities and/or issues of nested menus in improved accessibility!

**Comment by TylerAPfledderer** [comment]
@pettinarip Following up on this.

Looks like we are currently waiting on #8635 to be addressed and there is the `NakedButton` that is in the Nav that I haven't seen migrated yet (looks like @wackerow was going to do it in #6374). 

Should this issue be taken care of or still wait for the `Nav` to be available for migration and handle this at that point?

**Comment by corwintines** [comment]
@pettinarip @TylerAPfledderer I would be in favor of doing it now unless there's a reason to wait

**Comment by TylerAPfledderer** [comment]
@corwintines I'm assigned to handling the migration of Nav. 

Waiting on PR #9509 for the DocSearch to be merged and I think this could be taken care of with the nav migration in parallel with a second iteration of this DocSearch.

Or one at a time with either second iteration of DocSearch or the nav migration done next?
